HeterogeneousCore/CUDAServices/plugins/NVProfilerService.cc

```diff
@@ -502,7 +502,7 @@ void NVProfilerService::preallocate(edm::service::SystemBounds const& bounds) {
   // create the NVTX domains for per-EDM-stream transitions
   stream_domain_.resize(concurrentStreams);
   for (unsigned int sid = 0; sid < concurrentStreams; ++sid) {
-    stream_domain_[sid] = nvtxDomainCreate((boost::format("EDM Stream %d") % sid).str().c_str());
+    stream_domain_[sid] = nvtxDomainCreate(fmt::sprintf("EDM Stream %d", sid).c_str());
```

Comment:
Is it actually safe ?
Does the C string live long enough to be copied by the `nvtxDomainCreate` function ?

Of course this would have been an issue also before...

Reply:
It should be safe. From 15.2 of N4659 (last C++17 draft)

>  Temporary objects are destroyed as the last step in evaluating the full-expression (4.6) that (lexically) contains the point where they were created. This is true even if that evaluation ends in throwing an exception. The value computations and side effects of destroying a temporary object are associated only with the full-expression, not with any specific subexpression.